_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-search.php

```diff
@@ -0,0 +1,79 @@
+<?php
+/**
+ * Proxy endpoint for Jetpack Search
+ *
+ * @package Jetpack
+ */
+
+use Automattic\Jetpack\Connection\Client;
+
+/**
+ * Jetpack Search: Makes authenticated requests to the site search API using blog tokens.
+ * This endpoint will only be used when trying to search private Jetpack and WordPress.com sites.
+ *
+ * @since 8.10
+ */
+class WPCOM_REST_API_V2_Endpoint_Search extends WP_REST_Controller {
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		$this->namespace = 'wpcom/v2';
+		$this->rest_base = 'search';
+
+		add_action( 'rest_api_init', array( $this, 'register_routes' ) );
+	}
+
+	/**
+	 * Called automatically on `rest_api_init()`.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			$this->rest_base,
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_search_results' ),
+					'permission_callback' => 'is_user_logged_in',
```

Comment:
This is probably the best way to go, I just briefly wondered if we should use current_user_can( 'read' ). I suspect they are equivalent though. Probably a question for Crew.

Reply:
`is_user_logged_in` should do the trick here 👍